problem3-lowest-common-ancestor-tree.coffee

```diff
@@ -53,17 +31,14 @@
 # * @return {TreeNode}
 # */
 lowestCommonAncestor = (root, p, q) ->
-  recurse = (node) ->
-    return null if node is null
-    return node if p.val is node.val or q.val is node.val
-
-    left = recurse(node.left)
-    right = recurse(node.right)
+  return null if root is null
+  return root if p.val is root.val or q.val is root.val
 
-    if left isnt null and right isnt null
-      return node
-    else if left isnt null
-      return node.left
-    return node.right
+  left = lowestCommonAncestor(root.left, p, q)
+  right = lowestCommonAncestor(root.right, p, q)
 
-  recurse(root)
+  if left isnt null and right isnt null
+    return root
+  else if left isnt null
+    return root.left
```

Comment:
You have to return the left and right of the already calculated current node. They are stored in your `left` and `right` variables.

Reply:
Ah!!!  🙏🏽  Will try that right now!

Reply:
That did it thanks!